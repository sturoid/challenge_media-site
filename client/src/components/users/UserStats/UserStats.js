import React from 'react';
import PropTypes from 'prop-types';
import EyeOutline from 'react-icons/lib/ti/eye-outline';
import ThumbsUp from 'react-icons/lib/ti/thumbs-up';
import Message from 'react-icons/lib/ti/message';
import userStats from './UserStats.styles';

const UserStats = ({ stats }) => (
  <div className={userStats}>
    <div>
      <div>
        <EyeOutline className="icon" /> Project Views
      </div>
      <div>{stats.project_views}</div>
    </div>
    <div>
      <div>
        <ThumbsUp className="icon" /> Appreciations
      </div>
      <div>{stats.project_appreciations}</div>
    </div>
    <div>
      <div>
        <Message className="icon" /> Comments
      </div>
      <div>{stats.project_comments}</div>
    </div>
  </div>
);

UserStats.propTypes = {
  stats: PropTypes.shape({
    project_views: PropTypes.number.isRequired,
    project_appreciations: PropTypes.number.isRequired,
    project_comments: PropTypes.number.isRequired,
    profile_views: PropTypes.number.isRequired
  }).isRequired
};
export default UserStats;
