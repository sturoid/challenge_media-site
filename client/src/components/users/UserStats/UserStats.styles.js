import { css } from 'emotion';
import { greyMid } from '../../../lib/colors.styles';

const userStats = css`
  margin-top: 15px;
  > div {
    padding: 3px 0;
    display: flex;
    align-items: center;
    font-size: 13px;
    justify-content: space-between;
    & > div:first-child {
      display: flex;
      align-items: center;
    }
    .icon {
      color: ${greyMid};
      font-size: 20px;
    }
  }
`;

export default userStats;
