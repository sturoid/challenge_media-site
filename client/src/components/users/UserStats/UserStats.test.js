import React from 'react';
import renderer from 'react-test-renderer';
import UserStats from './UserStats';
import statsData from '../../../../../lib/test-data/userStats';

describe('<UserStats />', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer
      .create(<UserStats stats={statsData.all_time} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
