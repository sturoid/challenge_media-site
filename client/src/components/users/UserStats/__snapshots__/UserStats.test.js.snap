// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly and equals snapshot 1`] = `
<div
  className="css-jiwkox"
>
  <div>
    <div>
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m20 15c2 0 4 0.6 5.5 1.7 2.2 1.5 4 3.5 5.3 5-1.3 1.4-3.1 3.4-5.3 5-1.5 1-3.5 1.6-5.5 1.6s-4-0.6-5.5-1.6c-2.2-1.6-4-3.6-5.3-5 1.3-1.5 3.1-3.5 5.3-5 1.5-1.1 3.5-1.7 5.5-1.7z m0-3.3c-2.8 0-5.4 0.8-7.4 2.3-4.4 3-7.6 7.7-7.6 7.7s3.2 4.6 7.6 7.7c2 1.4 4.6 2.3 7.4 2.3s5.4-0.9 7.4-2.3c4.4-3.1 7.6-7.7 7.6-7.7s-3.2-4.7-7.6-7.7c-2-1.5-4.6-2.3-7.4-2.3z m0 8.3c-0.9 0-1.7 0.7-1.7 1.7 0 0.9 0.8 1.6 1.7 1.6 0.9 0 1.7-0.7 1.7-1.6 0-1-0.8-1.7-1.7-1.7z m0 6.7c-2.8 0-5-2.3-5-5s2.2-5 5-5 5 2.2 5 5-2.2 5-5 5z m0-8.4c-1.8 0-3.3 1.5-3.3 3.4s1.5 3.3 3.3 3.3 3.3-1.5 3.3-3.3-1.5-3.4-3.3-3.4z"
          />
        </g>
      </svg>
       Project Views
    </div>
    <div>
      33488
    </div>
  </div>
  <div>
    <div>
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m32.6 14.5c-0.6-0.3-4.2-0.7-6.5-1 0.4-2.1 0.6-4 0.6-6 0-2.3-1.9-4.2-4.2-4.2s-4.2 1.9-4.2 4.2c0 3.1-1.1 4.6-2.7 6.2-0.9-1.2-2.3-2-3.9-2-2.8 0-5 2.2-5 5v10c0 2.7 2.2 5 5 5 1.2 0 2.4-0.5 3.2-1.3 0.1 0.1 0.2 0.2 0.4 0.3 1.6 1.3 6.1 2.6 9.7 2.6 3.1 0 4.3-0.5 5.4-0.9 0.2-0.1 0.4-0.1 0.6-0.2 1.3-0.5 2.6-2 2.8-3.7l1.2-9.9c0.2-1.7-0.8-3.6-2.4-4.1z m-20.9 13.8c-1 0-1.7-0.7-1.7-1.6v-10c0-1 0.7-1.7 1.7-1.7s1.6 0.7 1.6 1.7v10c0 0.9-0.7 1.6-1.6 1.6z m18.8-0.2c0 0.4-0.4 0.9-0.7 1-0.2 0.1-0.4 0.1-0.6 0.2-0.9 0.4-1.7 0.7-4.2 0.7-3.2 0-6.9-1.3-7.7-1.9-0.3-0.2-0.6-1-0.6-1.4v-8.3c0-0.1 0.1-1.1 1.1-2.2 1.6-1.5 3.9-3.8 3.9-8.7 0-0.5 0.3-0.8 0.8-0.8s0.8 0.3 0.8 0.8c0 2.3-0.2 4.5-0.8 7.2l-0.5 2.3 2.2-0.3c1 0 6.6 0.7 7.3 0.9 0.1 0 0.2 0.3 0.1 0.5l-1.1 10z"
          />
        </g>
      </svg>
       Appreciations
    </div>
    <div>
      1841
    </div>
  </div>
  <div>
    <div>
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m30 11.7c0.9 0 1.7 0.7 1.7 1.6v11.7c0 0.9-0.8 1.7-1.7 1.7h-14.7l-0.3 0.3v-0.3h-5c-0.9 0-1.7-0.8-1.7-1.7v-11.7c0-0.9 0.8-1.6 1.7-1.6h20z m0-3.4h-20c-2.7 0-5 2.3-5 5v11.7c0 2.8 2.3 5 5 5h1.7v5l5-5h13.3c2.8 0 5-2.2 5-5v-11.7c0-2.7-2.2-5-5-5z"
          />
        </g>
      </svg>
       Comments
    </div>
    <div>
      20
    </div>
  </div>
</div>
`;
